andela-eshaibu/document_management_system
# Implement Feedback

#### What does this PR do?

This PR implements the feedback from checkpoint 2 defence
#### Description of Task to be completed?

The task implements ability for admin user to manage users (create, edit, delete and search). It also implements profile view and edit of logged in user
* Remove non descriptive naming keys in api response
* Test for actual values in tests
* Change controllers to pascal case and other variables to camel case
* Move check for owner of document in update and delete function to a middleware.
* Make one endpoint for admin and regular users to get documents accessible.
* Improve JS-doc for functions
* Move profile to top right of page
* Add limit for pagination
* Send email when admin creates user
* Write more test and improved test for reducers, components and end-2-end test

#### How should this be manually tested?
* Visit e-docman.herokuapp.com
* Clone the repo using : `git clone https://github.com/andela-eshaibu/document_management_system.git`
* Run the following command in your terminal to install dependencies
```
cd document_management_system
npm install
```
* Create a `.env` file in the repo's root directory, copy the content of the `.env.sample` file and  paste in `.env` file.
* Edit the .env file content with the appropriate value for your database url values and default information about the default admin.
* Run `npm start` to start application
* Open your browser and navigate to `http://localhost:8000`

#### Any background context you want to provide?
N/A

#### Screenshots (if appropriate)
N/A
